data_for_test.go

```diff
@@ -369,6 +369,23 @@ var suites = []FixtureSuite{
 			},
 		},
 	},
+        // exec false reports a span
+        // { 
+        //         {
+        //                 Name: "#258 Commands that exit with a non-zero exit code should report a span",
+        //                 Config: FixtureConfig{
+        //                         CliArgs: []string{"exec", "--endpoint", "{{endpoint}}", "--", "false"},
+        //                 },
+        //                 Expect: Results{
+        //                         SpanCount: 1,
+        //                         CliOutput: "",
+        //                         Diagnostics: otelcli.Diagnostics{
+        //                                 ExecExitCode: 1,
+        //                         },
+	// 			Config:    otelcli.DefaultConfig().WithEndpoint("grpc://{{endpoint}}"),
+        //                 },
+        //         },
+        // },
```

Comment:
Hmmm this test looks right.

Try adding `"--fail", "--verbose"` to the CliArgs. It looks like `config.SoftFail` is being called when child.Run() returns an err.

> If the command starts but does not complete successfully, the error is of type *ExitError. Other error types may be returned for other situations.

so the `if err :- child.Run(); err != nil` needs to be expanded to check for ExitError and not SoftFail on that so you can send the error upstream in the span status.